library/src/main/java/org/apache/apex/malhar/lib/state/managed/Bucket.java

```diff
@@ -454,6 +509,14 @@ public long freeMemory(long windowId) throws IOException
         entryIter.remove();
 
         for (Map.Entry<Slice, BucketedValue> entry : windowData.entrySet()) {
+          /**
+           * The data still in memory and reachable before the memory released
+           * So put key into bloom filter here
+           */
+          if (bloomFilter != null) {
```

Comment:
Adding an element to the bloom filter is best done when the element is fetched. This should also be in the first `get` call.